### PR TITLE
rest firmware put bugfix for when release_date is omitted

### LINKED ISSUE #687: REST API: Optional release_date is improperly sanitized
## Short description

FACT handles the absence of its optional release_date key during REST firmware upload improperly 

## Reproducer

```bash
xh put http://localhost:5000/rest/firmware device_name=device device_part=complete device_class=router file_name=fw.tar vendor=vendor version=1.0 binary="$(base64 fw.tar)"
```

## Traceback

```python
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/home/fact/FACT_core/src/helperFunctions/process.py", line 57, in run
    raise exception
  File "/home/fact/FACT_core/src/helperFunctions/process.py", line 52, in run
    Process.run(self)
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/fact/FACT_core/src/scheduler/Analysis.py", line 236, in scheduler
    self.process_next_analysis(task)
  File "/home/fact/FACT_core/src/scheduler/Analysis.py", line 251, in process_next_analysis
    self.pre_analysis(fw_object)
  File "/home/fact/FACT_core/src/storage/db_interface_backend.py", line 17, in add_object
    self.add_firmware(fo_fw)
  File "/home/fact/FACT_core/src/storage/db_interface_backend.py", line 67, in add_firmware
    entry = self.build_firmware_dict(firmware)
  File "/home/fact/FACT_core/src/storage/db_interface_backend.py", line 91, in build_firmware_dict
    'release_date': convert_str_to_time(firmware.release_date),
  File "/home/fact/FACT_core/src/helperFunctions/data_conversion.py", line 105, in convert_str_to_time
    return datetime.strptime(string, '%Y-%m-%d')
TypeError: strptime() argument 1 must be str, not None
```

### PR DESCRIPTION
**Pull request by jstucke**
- added default date (`1970-01-01`)
- added validation for release_date field

resolves #687

### CODE REVIEW

File: src/web_interface/rest/rest_firmware.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
     'file_name':  fields.String(description='File Name', required=True),
     'version':  fields.String(description='Version', required=True),
     'vendor':  fields.String(description='Vendor', required=True),
-    'release_date':  fields.String(description='Release Date'),
+    'release_date':  fields.String(description='Release Date', default='1970-01-01'),
```
Comment (rhelmke) [inline src/web_interface/rest/rest_firmware.py:32]: This works okay. But as we concluded, we should also validate the passed date string upon retrieval to keep the data nice and tidy

**Review by rhelmke — CHANGES_REQUESTED** [review]

File: src/web_interface/rest/rest_firmware.py (hunk 1 above)
Comment (jstucke) [reply src/web_interface/rest/rest_firmware.py]: added validation

Comment (rhelmke) [reply src/web_interface/rest/rest_firmware.py]: Sadly, the validation breaks the [test_rest_upload_valid](https://github.com/fkie-cad/FACT_core/blob/9da01ce172a7b817c9ab23aeb90afe3870f1e7eb/src/test/integration/web_interface/rest/test_rest_firmware.py#L61-L76) test, which now provides an invalid date value.

Comment (jstucke) [reply src/web_interface/rest/rest_firmware.py]: should be fixed now

**Review by rhelmke — APPROVED** [review]

**Pull request merged**